### DISCUSSION TITLE
Dynamically resize the caption height

### DISCUSSION THREAD

**Discussion by KailangHuo** [AvaloniaUI/Avalonia#14123] [Q&A]
Hello guys,  I'm using this ExtendClientAreaToDecorationsHint="True" to my window, and also use this piece of code to handle the maxmized-wrong-padding problem.
```
<Window.Styles>
        <Style Selector="Window[WindowState=Maximized]">
            <Setter Property="Padding" Value="8" />
        </Style>
    </Window.Styles>
```
I know its a win 32 problem, and the schemes over here generally reached all of my needs in my project. But there's still one.
How can I correctly resize the caption height of my window? this also includes the size of the caption buttons.

for example, the screenshots of the window when its normal size:
![image](https://github.com/AvaloniaUI/Avalonia/assets/88365925/9883d82e-0777-4f29-8c94-782b75c0f71f)
The caption height is also at this size, and when its maximized:
![image](https://github.com/AvaloniaUI/Avalonia/assets/88365925/d0e9dffd-63e9-476a-a8a2-8b471be9405d)
I hope I conveyed my question consicely.
Any help is appreciated!

**Comment by timunie** [comment]
simply use `Padding="{Binding $self.OffScreenMargin}"`

**Reply by KailangHuo** [reply]
to which element though? To the window?
this is my application:
```
    <Window.Styles>
        <Style Selector="Window[WindowState=Maximized]">
            <Setter Property="Padding" Value="{Binding $self.OffScreenMargin}" />
        </Style>
    </Window.Styles>
```

I also directly apply this piece of code to the mainwindow's property:
```
...
 FontWeight="Light"
 Icon="/Assets/avalonia-logo.ico"
  Padding="{Binding $self.OffScreenMargin}"
...
```

and its still wrong. 
Could you please give more information. Thx

**Reply by timunie** [reply]
I've applied this Padding to my first child control. There was no need for any Selector for Maximized as this will update automatically. If you have  a custom title bar, make sure it is affected by this. Check also DevTools [F12]. 

If your issue is not solved, upload a minimal sample or complete XAML of the Window, not just parts of it.
